src/util/transaction_test_util.cpp

```diff
@@ -0,0 +1,265 @@
+#include <utility>
+#include <algorithm>
+#include <vector>
+#include "util/transaction_test_util.h"
+
+namespace terrier {
+RandomWorkloadTransaction::RandomWorkloadTransaction(LargeTransactionTestObject *test_object)
+    : test_object_(test_object),
+      txn_(test_object->txn_manager_.BeginTransaction()),
+      aborted_(false),
+      start_time_(txn_->StartTime()),
+      commit_time_(UINT64_MAX),
+      buffer_(test_object->bookkeeping_ ? nullptr : new byte[test_object->row_size_]) {}
+
+RandomWorkloadTransaction::~RandomWorkloadTransaction() {
+  if (!test_object_->gc_on_) delete txn_;
+  if (test_object_->bookkeeping_) delete[] buffer_;
+  for (auto &entry : updates_)
+    delete[] reinterpret_cast<byte *>(entry.second);
+  for (auto &entry : selects_)
+    delete[] reinterpret_cast<byte *>(entry.second);
+}
+template<class Random>
```

Comment:
It's not clear to me why this works in a .cpp without explicit instantiation but sure!